channeldb/invoices.go

```diff
@@ -650,6 +670,46 @@ func (d *DB) SettleInvoice(paymentHash [32]byte,
 	return settledInvoice, err
 }
 
+// SettleHoldInvoice sets the preimage of a hodl invoice and marks the invoice
+// as settled.
+func (d *DB) SettleHoldInvoice(preimage lntypes.Preimage) (*Invoice, error) {
+	var updatedInvoice *Invoice
+	hash := preimage.Hash()
+	err := d.Batch(func(tx *bbolt.Tx) error {
```

Comment:
qq: did this use `Batch` previously? i think we want to set `updatedInvoice` to nil during each invocation to avoid side-effects reexecution of batch updates

Reply:
It was `Update` before, but iirc it was your own suggestion to use `Batch`. But now I actually reverted it back to `Update`. I don't think there is a practical advantage of `Batch` here, only more things that can go wrong.

Reply:
was thinking the same thing, can always revisit if there is evidence that this is a bottleneck